DopplerCustomDomain/Controllers/CustomDomainController.cs

```diff
@@ -26,5 +27,26 @@ public class CustomDomainController
             _customDomainProviderService = customDomainProviderService;
             _serviceNameResolver = serviceNameResolver;
         }
+
+        [HttpPost("/{service}/{domain}")]
+        public async Task<IActionResult> CreateCustomDomain(string service, string domain)
+        {
+            var serviceName = _serviceNameResolver.Resolve(service);
+
+            if (string.IsNullOrEmpty(serviceName))
+            {
+                return new NotFoundObjectResult($"Cannot find the service called: {service}");
+            }
+
+            try
+            {
+                await _customDomainProviderService.CreateCustomDomain(serviceName, domain);
+                return new OkObjectResult("Custom Domain Created");
+            }
+            catch (Exception ex)
+            {
+                return new BadRequestObjectResult(ex);
```

Comment:
Is it a bad request? In my opinion, it is more like an internal server error...

You can keep the exception without catching and allow Problem Details middleware to deal with it.